src/OpenTelemetry.Collector.Dependencies/Implementation/HttpHandlerDiagnosticListener.cs

```diff
@@ -45,9 +48,11 @@ public HttpHandlerDiagnosticListener(Tracer tracer, HttpClientCollectorOptions o
             // Depending on the .NET version/flavor this will look like
             // '.NETCoreApp,Version=v3.0', '.NETCoreApp,Version = v2.2' or '.NETFramework,Version = v4.7.1'
 
-            if (framework != null && framework.Contains("Version=v3"))
+            if (framework != null)
             {
-                this.httpClientSupportsW3C = true;
+                var match = CoreAppMajorVersionCheckRegex.Match(framework);
+
+                this.httpClientSupportsW3C = match.Success && int.Parse(match.Groups[1].Value) >= 3;
```

Comment:
why is that better than contains?

Reply:
What I wanted to do was future-proof the code. First I did something like this:
```csharp
if (framework != null && framework.Contains(".NETCoreApp,") && (framework.Contains("Version=v3") || framework.Contains("Version=v5"))
```
And then I pictured someone else having to revisit it in the future...
```csharp
if (framework != null && framework.Contains(".NETCoreApp,") && (framework.Contains("Version=v3") || framework.Contains("Version=v5") || framework.Contains("Version=v6"))
```
The Regex seemed like a better way to express what we are trying to accomplish... .NET Core major version >= 3.

Reply:
I don't know if supporting future versions of .NET makes sense, but don't have any strong opinion about this.